### DISCUSSION TITLE
How to disabled css-in-js?

### DISCUSSION THREAD

**Discussion by realyuki** [ant-design/ant-design#51704] [Q&A]
I'm migrating from v3 to v5. 

I was in trouble because the style was being processed in js and the style was broken.

How do I invalidate v5's css-in-js?

**Comment by petercat-assistant** [comment]
在 Ant Design v5 中，默认使用了 CSS-in-JS 来处理样式。如果你希望禁用 CSS-in-JS，可以通过以下步骤来实现：

1. **禁用 CSS-in-JS**: 你可以通过配置来禁用 CSS-in-JS。具体来说，可以在项目的配置文件中设置 `css-in-js` 为 `false`。具体的配置方式可能会根据你使用的构建工具（如 Webpack、Vite 等）有所不同。

2. **使用传统的 CSS 文件**: 确保在禁用 CSS-in-JS 后，你的项目中仍然引入了 Ant Design 的 CSS 文件。你可以在项目的入口文件中手动引入 Ant Design 的 CSS 文件，例如：
   ```javascript
   import 'antd/dist/antd.css';
   ```

3. **检查样式覆盖**: 如果你在 v3 中有自定义样式，确保这些样式在 v5 中仍然有效。你可能需要根据 v5 的新样式结构调整自定义样式。

如果你在禁用 CSS-in-JS 的过程中遇到问题，欢迎在评论中详细描述你的问题，并 @petercat-assistant 来与我进行进一步的交流。

**Comment by zombieJ** [comment]
v5 use cssinjs. What's your `broken` mean?

**Comment by michael-vasyliv** [comment]
I have shitty performance because of css-in-js I use Typography.Text to render cell in my custom table and performance terrible, 1 sec to render 10 rows with 10 cell in each row.
Any ideas how to remove css-in-js from "antd": "^5.22.2"?

**Reply by michael-vasyliv** [reply]
with Typography.Text
[Screencast from 02.01.25 19:40:50.webm](https://github.com/user-attachments/assets/109bf5f9-8224-4aa9-94f1-b1f28196efba)
without Typography.Text
[Screencast from 02.01.25 19:41:24.webm](https://github.com/user-attachments/assets/5b77c864-d3ae-4654-a8aa-0ca9acbd8cab)
